### DISCUSSION TITLE
Generate preview image of PDF

### DISCUSSION THREAD

**Discussion by Whip** [dompdf/dompdf#2467] [General]
I'm surprised this hasn't been asked yet (or I couldn't find it). I wanted to know if its possible for dompdf to generate a preview image of the pdf after its generated it from HTML?

**Comment by sirbenson** [comment]
Print the pdf directly to the browser: $dompdf->stream("filename.pdf", array("Attachment" => false));

**Comment by Whip** [comment]
That's not an image. And that's certainly not an efficient way to load thumbnails.

**Comment by bsweeney** [comment]
Dompdf directly could not generate a preview image based on the PDF but Dompdf could generate an image render of the HTML. It would be full size, so depending on your purpose you would still have to perform some manipulation of the image.

**Reply by Whip** [reply]
Could you please tell me how that can be done?

**Reply by bsweeney** [reply]
To generate an image you can use the GD adapter instead of the CPDF or PDFLib adapter. The results may be slightly different due to the different rendering engine, but should be nearly identical.

```
$dompdf = new Dompdf(['pdfBackend' => 'GD'])
$dompdf->loadHtml($html);
$dompdf->render();
$dompdf->stream(); // or output to capture the image contents in a variable
```
